parts/kubernetesmastercustomscript.sh

```diff
@@ -92,6 +93,43 @@ function ensureKubectl() {
     fi
 }
 
+function setNetworkPlugin () {
+    sed -i "s/^KUBELET_NETWORK_PLUGIN=.*/KUBELET_NETWORK_PLUGIN=${1}/" /etc/default/kubelet
+}
+
+function setDockerOpts () {
+    sed -i "s#^DOCKER_OPTS=.*#DOCKER_OPTS=${1}#" /etc/default/kubelet
+}
+
+function configNetworkPolicy() {
+    if [[ ! -z "${APISERVER_PRIVATE_KEY}" ]]; then
+        # on masters
+        ADDONS="calico-configmap.yaml calico-daemonset.yaml"
+        ADDONS_PATH=/etc/kubernetes/addons
+        CALICO_URL="https://raw.githubusercontent.com/projectcalico/calico/a4ebfbad55ab1b7f10fdf3b39585471f8012e898/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend-addon-manager"
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                curl -sSL --retry 12 --retry-delay 10 "${CALICO_URL}/${addon}" > "${ADDONS_PATH}/${addon}"
+            done
+        else
+            # download calico yamls
+            for addon in ${ADDONS}; do
+                rm -f "${ADDONS_PATH}/${addon}"
+            done
+        fi
+    else
+        # on agents
+        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
```

Comment:
This may be confusion on my part due to unfamiliarity with Calico, but is `NETWORK_POLICY` the same as `CNI_PLUGIN` in a sense?

For example, if I knew of another branch somewhere that is adding another CNI plugin, would they simply be able to add their plugin as a new option for `NETWORK_POLICY`, here and in the apimodel? It seems like it, but I wanted to check.